src/dimension_slice.c

```diff
@@ -114,23 +114,34 @@ lock_result_ok_or_abort(TupleInfo *ti, DimensionSlice *slice)
 		case TM_Deleted:
 			ereport(ERROR,
 					(errcode(ERRCODE_LOCK_NOT_AVAILABLE),
-					 errmsg("dimension slice %d deleted by other transaction", slice->fd.id),
+					 errmsg("chunk deleted by other transaction"),
 					 errhint("Retry the operation again.")));
 			pg_unreachable();
 			break;
 #endif
 		case TM_Updated:
-			ereport(ERROR,
-					(errcode(ERRCODE_LOCK_NOT_AVAILABLE),
-					 errmsg("dimension slice %d locked by other transaction", slice->fd.id),
-					 errhint("Retry the operation again.")));
+			/* If the tid and ctid in the lock failure data is the same, then
+			 * this is a delete. Otherwise it is an update and ctid is the new
+			 * tuple ID. This applies mostly to PG11, since PG12 has an
+			 * explicit lockresult for delete tuples. */
+			if (memcmp(ts_scanner_get_tuple_tid(ti), &ti->lockfd.ctid, sizeof(ti->lockfd.ctid)) ==
+				0)
+				ereport(ERROR,
+						(errcode(ERRCODE_LOCK_NOT_AVAILABLE),
+						 errmsg("chunk deleted by other transaction"),
+						 errhint("Retry the operation again.")));
+			else
+				ereport(ERROR,
```

Comment:
In practice, this case should never occur. Dimension slices are immutable so they should never be updated. But I am not sure we are confident enough to assert it here.